### PR TITLE
feat: add css function in web builder

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details
 css plugin for web builder.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/css-config/src/postcss.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const getPostcssConfig = (
     },
   };
 
-  const enableCssMinify = config.output?.disableCssExtract && isProd();
+  const enableCssMinify = config.output?.disableMinimize && isProd();
```
Comment (chenjiahan) [inline packages/cli/css-config/src/postcss.ts:28]: 这里原来写的 `config.output?.disableCssExtract` 需要保留一下

背景：

- 如果 css 抽取为独立文件，那么会触发 css-minimizer-plugin 压缩，不需要额外配置 cssnano
- 如果 css 走 style-loader，那么不会触发 css-minimizer-plugin，需要通过 cssnano 来压缩

File: packages/builder/web-builder/src/plugins/css.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          {},
+          config.tools?.styleLoader || {},
+        );
+        const cssLoaderOptions = applyOptionsChain<CSSLoaderOptions, null>(
+          {
+            importLoaders: 1,
+            modules: {
+              exportLocalsConvention: 'camelCase',
+            },
+            sourceMap: enableSourceMap,
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/plugins/css.ts:51]: eden 和 modern 的 cssLoader 默认值应该差别挺大的，可以再详细对比下

主要是 localIdentName、exportOnlyLocals、exportLocalsConvention 这几个字段

File: packages/cli/css-config/src/postcss.ts (hunk 1 above)
Comment (sanyuan0704) [reply packages/cli/css-config/src/postcss.ts:28]: 那这个包我先不动了

File: packages/builder/web-builder/src/plugins/css.ts (hunk 2 above)
Comment (sanyuan0704) [reply packages/builder/web-builder/src/plugins/css.ts:51]: 两者的 css loader 的 modules 配置只有 localIdentName 不一样，其它的默认参数我都放进来，localIdentName 的话沿用 Eden 的:
```ts
// eden 
localIdentName: BUILD ? '[hash:base64]' : '[path][name]__[local]--[hash:base64:5]',
// modern
 localIdentName: config.output
    ? config.output.cssModuleLocalIdentName!
    : '',
```

Comment (chenjiahan) [reply packages/builder/web-builder/src/plugins/css.ts:51]: 可以滴

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
后续可以再提个 PR 把相关依赖预打包一下

https://github.com/modern-js-dev/modern.js/tree/main/scripts/prebundle

**Pull request merged**